src/js/components/MaskedInput/MaskedInput.js

```diff
@@ -203,9 +203,14 @@ const MaskedInput = forwardRef(
         // Align with the mask.
         const nextValueParts = parseValue(mask, event.target.value);
         const nextValue = nextValueParts.map(part => part.part).join('');
-        if (value !== nextValue) setInputValue(nextValue);
-        if (onChange) onChange(event);
-        setValue(nextValue);
+        // eslint-disable-next-line no-param-reassign
+        event.target.value = nextValue;
+
+        if (value !== nextValue) {
+          setInputValue(nextValue);
+          if (onChange) onChange(event);
+          setValue(nextValue);
+        }
```

Comment:
I like the direction to not notify if the value isn't being changed. But, I'm hesitant to reach into the event object and modify it. I think if we remove lines 206 and 207, it will still behave correctly. Can we try that?

Reply:
Oh wow, that totally worked, thanks for the suggestion! I wasn't really happy with those lines, but I thought they were necessary haha.

I'll make a commit to remove those.